app/src/components/widgets/charts/cumulativeTrendChart/cumulativeTrendChart.jsx

```diff
@@ -0,0 +1,242 @@
+          categories: this.dataGroupNames.map((category) => {
+            const prefix = widget.contentParameters.widgetOptions.prefix;
+            return category.indexOf(prefix) > -1 ? category.split(`${prefix}:`)[1] : category;
+          }),
+          tick: {
+            centered: true,
+            inner: true,
+          },
+        },
+        y: {
+          show: !isPreview,
+        },
+      },
+      grid: {
+        y: {
+          show: !isPreview,
+        },
+      },
+      size: {
+        height: this.height,
+      },
+      interaction: {
+        enabled: !isPreview,
+      },
+      padding: {
+        top: isPreview ? 0 : 85,
+      },
+      legend: {
+        show: false, // we use custom legend
+      },
+      tooltip: {
+        grouped: false,
+        position: this.getPosition,
+        contents: this.renderContents,
+      },
+    };
+
+    this.setState({
+      isConfigReady: true,
+    });
+  };
+
+  getPosition = (d, width, height) => {
+    const rect = this.node.getBoundingClientRect();
+    const top = this.y - rect.top - height;
+    let left = this.x - rect.left - width / 2;
+
+    if (left < 0) {
+      left = 0;
+    } else if (left + width > rect.width) {
+      left = rect.width - width;
+    }
+
+    return {
+      top: top - 8,
+      left,
+    };
+  };
+
+  getCoords = ({ pageX, pageY }) => {
+    this.x = pageX;
+    this.y = pageY;
+  };
+
+  resizeChart = () => {
+    const newHeight = this.props.container.offsetHeight;
+    const newWidth = this.props.container.offsetWidth;
+
+    if (this.height !== newHeight) {
+      this.chart.resize({
+        height: newHeight,
+      });
+      this.height = newHeight;
+    } else if (this.width !== newWidth) {
+      this.chart.flush();
+      this.width = newWidth;
+    }
+  };
+
+  renderContents = (d) => {
+    const index = d[0].index;
+    const groupName = this.dataGroupNames[index];
+    const columns = this.config.data.columns;
+    const itemsData = [];
+    columns.forEach((column) => {
+      const id = column[0];
+      const message = messages[id];
+      if (message) {
+        const value = column[index + 1] || 0;
+        itemsData.push({
+          id,
+          color: getColorForKey(id),
+          name: this.props.intl.formatMessage(message),
+          value,
+        });
+      }
+    });
+    return ReactDOMServer.renderToStaticMarkup(
+      <CumulativeTrendTooltip groupName={groupName} itemsData={itemsData} />,
+    );
+  };
+
+  render() {
+    const { isPreview } = this.props;
+    const classes = cx('cumulative-trend-chart', {
+      'preview-view': isPreview,
+    });
+    return (
+      <div className={classes}>
+        {this.state.isConfigReady && (
+          <C3Chart config={this.config} onChartCreated={this.onChartCreated}>
+            {!this.props.isPreview && (
+              <Legend
+                items={Object.getOwnPropertyNames(this.config.data.colors)}
+                onClick={this.onClick}
```

Comment:
probably descriptive method names will be more readable? for example `onClick={this.toggleChart}, onMouseOver={this.focusChart}`

Reply:
If we change logic of event handler in future do we need to update method name to something more appropriate like onClick={this.toggleEntireWidget} ?

Reply:
I didn't get for sure what do you mean, but my point is that function name usually reflects what it does, not when it is called. If the logic changes the name will probably change. The names I've provided are just to illustrate what I mean, because I don't know for sure what `this.chart.toggle(id)`

Reply:
I encourage descriptive names for methods as well and I understood your point but I disagree with your approach in this case. And I provided method name just for example either. I prefer to distinguish event handlers like onClick, onSubmit,.. from class methods with  business logic so you can change it anytime without updating a template. In another words onClick is always onClick and it doesn't depend on what do we do on it.